jina/executors/indexers/keyvalue.py

```diff
@@ -30,13 +30,17 @@ def __init__(self, path, mode):
 
         def close(self):
             """Close the file."""
-            self.body.close()
-            self.header.close()
+            if not self.body.closed:
```

Comment:
Is it possible that self.body.closed is true? But actually, it's not closed yet?
Like something else set this to true. Or it's not setable.

Reply:
I don't think so

```
In [1]: f = open('a','w')

In [2]: f.closed
Out[2]: False

In [3]: f.closed = True
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-dcb2d4390ee6> in <module>
----> 1 f.closed = True

AttributeError: attribute 'closed' of '_io.TextIOWrapper' objects is not writable

```